File: hw/cañarte/u3/HW30Observer&Strategy/Python/ObserverPython/controller/Investor.py
import sys
import os
# Saray Cañarte, codecreators, DCCO - ESPE

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from controller.IInvestor import IInvestor


class Investor(IInvestor):
    def __init__(self, name):
        self._name = name
        self._observer_state = None
        self._stock = None

    def update(self, stock, args):
        print(f"Notified observer {self._name}:")
        if isinstance(args, str):
            print(f"The symbol of {stock.symbol} changed to: {args}")
        elif isinstance(args, float):
            print(f"The price of {stock.symbol} changed to: {args}")

File: hw/CEDENOR/U3/HW30Observer&Strategy/Java/ObserverPatttern/src/main/java/ec/edu/espe/observerpattern/controller/Investor.java

package ec.edu.espe.observerpattern.controller;

/**
 *
 * @author Nahomi Cedeño, codecreators, DCCO - ESPE
 */
public class Investor implements IInvestor {
    private String name;
    private String observerState;
    private Stock stock;

    public Investor(String name) {
        this.name = name;
    }

    public void update(Stock stock, Object args) {
        System.out.println("Notified observer " + name + ": ");
        if (args instanceof String) {
            System.out.println("The symbol of " + stock.getSymbol() + " changed to: " + args);
        } else if (args instanceof Double) {
            System.out.println("The price of " + stock.getSymbol() + " changed to: " + args);
        }
    }
}


